server/svix-server/src/queue/mod.rs

```diff
@@ -40,11 +40,67 @@ impl MessageTask {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MessageTaskBatch {
+    pub msg_id: MessageId,
+    pub app_id: ApplicationId,
+    pub endpoint_ids: Vec<EndpointId>,
+    pub trigger_type: MessageAttemptTriggerType,
+    pub attempt_count: u16,
+}
+
+impl MessageTaskBatch {
+    pub fn new_task(
+        msg_id: MessageId,
+        app_id: ApplicationId,
+        endpoint_ids: Vec<EndpointId>,
+        trigger_type: MessageAttemptTriggerType,
+    ) -> QueueTask {
+        QueueTask::MessageBatchV1(Self {
+            msg_id,
+            app_id,
+            endpoint_ids,
+            attempt_count: 0,
+            trigger_type,
+        })
+    }
+    pub fn to_message_task(self, endpoint_id: &EndpointId) -> MessageTask {
```

Comment:
I'm a bit suspicious of this function. Not sure why it's needed.

Reply:
This is ultimately so that I can reuse `dispatch` for both `MessageTask` and `MessageTaskBatch` by always exploding `MessageTaskBatch` into multiple `MessageTask`s per endpoint, and executing those concurrently in `dispatch`. To me, the logic for that conversion felt more natural implemented as a method on `MessageTaskBatch` than constructed ad-hoc somewhere in the worker.

Reply:
But with your feedback of removing the endpoint_ids field completely and calculating them at work-time, this function probably becomes obsolete and can be removed.

Reply:
I didn't mean at "work-time" in the sense that it has to be in `dispatch` but rather that you call the logic that does the math in the worker (after the queue), rather than before.